bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/dto/CloudProviderDTO.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal.dto;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link CloudProviderDTO} class contains the information
+ * to allow encryption and decryption for the supported Cloud Providers
+ *
+ * @author Jacek Dobrowolski - Initial Contribution
+ * @author Bob Eckhoff - JavaDoc and conversion to record
+ */
+@NonNullByDefault
+public record CloudProviderDTO(String name, String appkey, String appid, String apiurl, String signkey, String proxied,
+        String iotkey, String hmackey) {
+
+    /**
+     * Cloud provider information
+     * All providers use the same signkey for AES encryption and Decryption.
+     * V2 Devices do not require a Cloud Provider entry as they only use AES
+     * 
+     * @param name Cloud provider
+     * @param appkey application key
+     * @param appid application id
+     * @param apiurl application url
+     * @param signkey sign key for AES
+     * @param iotkey iot key - MSmarthome only
+     * @param hmackey hmac key - MSmarthome only
+     * @param proxied proxy - MSmarthome only
+     * @param name Cloud provider
```

Comment:
This javadoc seems like a mix for the record and for the public method. They should each have their own javadoc with params that belong to their signatures.

Reply:
Made changes.  Not an expert on javaDoc.  I have depended on what pops up on VSC, but nothing was popping up for the record.  Please review.

Reply:
all good